src/main/java/com/artipie/api/artifactory/GetPermissionSlice.java

```diff
@@ -65,12 +70,27 @@ public GetPermissionSlice(final Settings settings) {
     public Response response(final String line, final Iterable<Map.Entry<String, String>> headers,
         final Publisher<ByteBuffer> body) {
         final Optional<String> opt = new FromRqLine(line, FromRqLine.RqPattern.REPO).get();
-        return opt.<Response>map(
-            repo -> new AsyncResponse(
-                new RepoPermissions.FromSettings(this.settings).permissions(repo)
-                    .thenApply(map -> new RsJson(GetPermissionSlice.response(map, repo)))
-            )
-        ).orElse(new RsWithStatus(RsStatus.BAD_REQUEST));
+        return new AsyncResponse(
+            opt.map(
+                repo -> new Unchecked<>(this.settings::storage).value()
+                    .exists(new Key.From(String.format("%s.yaml", repo))).thenCompose(
+                        exists -> {
+                            final CompletionStage<Response> res;
+                            if (exists) {
+                                res = new RepoPermissions.FromSettings(this.settings)
+                                    .permissions(repo).thenApply(
+                                        perms -> new RsJson(
+                                            GetPermissionSlice.response(perms, repo)
+                                        )
+                                    );
+                            } else {
+                                res = CompletableFuture.completedStage(StandardRs.NOT_FOUND);
+                            }
+                            return res;
+                        }
+                    )
+            ).orElse(CompletableFuture.completedFuture(new RsWithStatus(RsStatus.BAD_REQUEST)))
+        );
```

Comment:
@olenagerasimova we may wrap only the code in `opt.map(...)` with `AsyncResponse` and also return `RsWithStatus` without wrapping it into `CompletableFuture` in `.orElse(...)`. This will simplify code a little bit